p2p/enode/iter.go

```diff
@@ -215,11 +216,13 @@ func (m *FairMix) Next() bool {
 		case n, ok := <-source.next:
 			if ok {
 				m.cur = n
+				source.timeout = m.timeout
 				return true
 			}
 			// This source has ended.
 			m.deleteSource(source)
 		case <-timeout:
+			source.timeout /= 2
```

Comment:
You probably want to cap this 100ms so it doesn't go down fully to zero, otherwise 1 (total) unresponsive source could cause a spinloop.